modules/code-editor/src/backend/editor.ts

```diff
@@ -0,0 +1,130 @@
+import * as sdk from 'botpress/sdk'
+import fs from 'fs'
+import _ from 'lodash'
+import path from 'path'
+
+import { Config } from '../config'
+
+import { EditableFile, FileType, TypingDefinitions } from './typings'
+
+const FILENAME_REGEX = /^[0-9a-zA-Z_\-.]+$/
+
+export default class Editor {
+  private bp: typeof sdk
+  private _botId: string
+  private _typings: TypingDefinitions
+  private _config: Config
+
+  constructor(bp: typeof sdk, botId: string, config: Config) {
+    this.bp = bp
+    this._botId = botId
+    this._config = config
+  }
+
+  async fetchFiles() {
+    return {
+      actionsGlobal: this._config.allowGlobal && (await this._loadFiles('/actions', 'action')),
+      actionsBot: await this._loadFiles('/actions', 'action', this._botId)
+    }
+  }
+
+  async _validateContent({ name, botId, type }: Partial<EditableFile>) {
```

Comment:
not an issue at the moment but this doesn't really validate the content of the file. It's a metadata validator. We might want to rename this method accordingly